model/patch/patch.go

```diff
@@ -321,15 +321,32 @@ func (p *Patch) SetActivation(activated bool) error {
 
 // UpdateModulePatch adds or updates a module within a patch.
 func (p *Patch) UpdateModulePatch(modulePatch ModulePatch) error {
+	// update the in-memory patch
+	patchFound := false
+	for i, patch := range p.Patches {
+		if patch.ModuleName == modulePatch.ModuleName {
+			p.Patches[i] = modulePatch
+			patchFound = true
+			break
+		}
+	}
+	if !patchFound {
+		p.Patches = append(p.Patches, modulePatch)
+	}
+	if modulePatch.Message != "" {
+		p.Description = modulePatch.Message
+	}
+
 	// check that a patch for this module exists
 	query := bson.M{
 		IdKey:                                 p.Id,
 		PatchesKey + "." + ModulePatchNameKey: modulePatch.ModuleName,
 	}
-	update := bson.M{
-		"$set": bson.M{PatchesKey + ".$": modulePatch},
+	update := bson.M{PatchesKey + ".$": modulePatch}
+	if modulePatch.Message != "" {
+		update[DescriptionKey] = modulePatch.Message
```

Comment:
Could p.Description already be set to something at this point? If not, do we need to consider `if modulePatch.Message != ""` since that's already a 0 value--couldn't we just set it?

Reply:
In the case of a regular patch, the patch's original message may or may not be set, but the modulePatch.Message will be "". If we wouldn't check the modulePatch.Message is not the empty string, adding a module to an existing patch would clear the patch's description.